jaxrs-clients/src/main/java/com/palantir/remoting/jaxrs/FeignJaxRsClientBuilder.java

```diff
@@ -0,0 +1,153 @@
+                .client(target.wrapClient(createOkHttpClient(userAgent)))
+                .retryer(target)
+                .options(createRequestOptions())
+                .logger(new Slf4jLogger(Client.class))
+                .logLevel(Logger.Level.BASIC)
+                .requestInterceptor(UserAgentInterceptor.of(userAgent))
+                .target(target);
+    }
+
+    private <T> FailoverFeignTarget<T> createTarget(Class<T> serviceClass, List<String> uris) {
+        return new FailoverFeignTarget<>(uris, serviceClass, backoffStrategy);
+    }
+
+    private Contract createContract() {
+        return new SlashEncodingContract(
+                new GuavaOptionalAwareContract(new JaxRsWithHeaderAndQueryMapContract()));
+    }
+
+    private Request.Options createRequestOptions() {
+        return new Request.Options(
+                (int) getThisConnectTimeout().or(CONNECT_TIMEOUT).getMillis(),
+                (int) getThisReadTimeout().or(READ_TIMEOUT).getMillis());
+    }
+
+    private Encoder createEncoder(ObjectMapper objectMapper) {
+        // TODO(rfink) Get Jackson24Encoder if ObjectMapper#writerFor is unavailable
+        return new InputStreamDelegateEncoder(new TextDelegateEncoder(new JacksonEncoder(objectMapper)));
+    }
+
+    private Decoder createDecoder(ObjectMapper objectMapper) {
+        // TODO(rfink) Get Jackson24Decoder if ObjectMapper#writerFor is unavailable
+        return new OptionalAwareDecoder(
+                new InputStreamDelegateDecoder(new TextDelegateDecoder(new JacksonDecoder(objectMapper))));
+    }
+
+    private feign.Client createOkHttpClient(String userAgent) {
+        okhttp3.OkHttpClient.Builder client = new okhttp3.OkHttpClient.Builder();
+        if (getThisSslSocketFactory().isPresent()) {
+            client.sslSocketFactory(getThisSslSocketFactory().get());
+        }
+
+        // Set up Zipkin/Brave tracing
+        ClientTracer tracer = ClientTracer.builder()
+                .traceSampler(Sampler.ALWAYS_SAMPLE)
+                .randomGenerator(new Random())
+                .state(new ThreadLocalServerClientAndLocalSpanState(
+                        getIpAddress(), 0 /** Client TCP port. */, userAgent))
+                .spanCollector(new SlfLoggingSpanCollector("ClientTracer(" + userAgent + ")"))
+                .build();
+        BraveOkHttpRequestResponseInterceptor braveInterceptor =
+                new BraveOkHttpRequestResponseInterceptor(
+                        new ClientRequestInterceptor(tracer),
+                        new ClientResponseInterceptor(tracer),
+                        new DefaultSpanNameProvider());
+        client.addInterceptor(braveInterceptor);
+
+        return new OkHttpClient(client.build());
+    }
+
+    // Returns the IP address returned by InetAddress.getLocalHost(), or -1 if it cannot be determined.
+    // TODO(rfink) What if there are multiple? Can we find the "correct" one?
+    private static int getIpAddress() {
```

Comment:
think it's pretty hard for users to determine their IP address. would probably prefer to just not set it or set it to 0.0.0.0 or something.